modules/python/python-logmsg.c

```diff
@@ -250,6 +250,31 @@ _logmessage_get_keys_method(PyLogMessage *self)
   return keys;
 }
 
+static PyObject *
+py_log_message_get(PyLogMessage *self, PyObject *args, PyObject *kwrds)
+{
+  const gchar *key = NULL;
+  Py_ssize_t key_len = 0;
+  PyObject *default_value = NULL;
+
+  static const gchar *kwlist[] = {"key", "default", NULL};
+  if (!PyArg_ParseTupleAndKeywords(args, kwrds, "z#|O", (gchar **) kwlist, &key, &key_len, &default_value))
+    return NULL;
+
+  PyObject *value = _get_value(self, key, self->cast_to_bytes);
+  if (!value)
+    return NULL;
+
+  if (value != Py_None)
+    return value;
+
+  if (!default_value)
+    return value; /* returns None with the proper refcount */
+
+  Py_XINCREF(default_value);
+  return default_value;
```

Comment:
`Py_None` is a valid value for an NV pair of type `LM_VT_NULL`. I think we shouldn't fallback to `default_value` in this case.

This is the case for python `dict`s as well:
```python
>>> d = {"a": None}
>>> print(d["a"])
None
>>> print(d.get("a"))
None
>>> print(d.get("a", "fallback"))
None
>>> print(d["b"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'b'
>>> print(d.get("b"))
None
>>> print(d.get("b", "fallback"))
fallback
```

Reply:
I'll take a look at this. My impression was that the LogMessage API does not make difference between NULL and unset values. There are signs of it, I was wrong.

Reply:
I've fixed this. The code looks even better now.